core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,90 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = Integer.MIN_VALUE;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-default", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The icon class specification e.g. 'category-icon-folder', 'category-icon-default', etc.
+     *
+     * @return the icon class specification
+     */
+    public abstract String getIconClassName();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
```

Comment:
It sounds like you guys are arguing about conflict here between having a clean (non-UI-specific) API and meeting the needs of what the UI would like to do with categories.

Goal: The UI would like to have the ability to let the category decide how may items need to be in a category before bothering to render it.  The thinking here is that the UI doesn't want to force conceptual complexity on the user if there aren't multiple item types in the category....however SOME categories may want to the themselves to be displayed even if there is only one item in the category.   This level of flexibility cannot exist only in the UI, therefore there needs to be a way for the category to communicate this behavior to a UI that can take advantage of this preference.  

* Alternative-1 - Provide default for this value (say 2), so that not all categories need to provide this). 
* Alternative-2, expose a bag of arbitrary attributes (name/value pairs) that can be used to propagate information to a consumer that supports it.  In this case, the NewItem UI could declare that it supports optional attributes like minToShow=1, which categories are able to use for finder-grained control when displaying the category in the Jenkins UI.
* Alternative-3 - move getMinToShow() interface method into a UICategory interface so that only categories that want this level of control in the UI need to override it.